client/question/domain/CodeEvalResultObjectFactory.js

```diff
@@ -174,6 +202,79 @@ tie.factory('CodeEvalResultObjectFactory', [
       return null;
     };
 
+    /**
+     * Compares the results to the expected values of the test cases, and
+     * returns an object containing task, testSuite, testCase index
+     * numbers and overall test number (0-indexed) of the first failed test
+     * case. The overall test number is defined by concatenating all the
+     * test cases together, in order.
+     *
+     * @param {Array<Task>} tasks The list of tasks for the current question.
+     * @returns {*|null} An object containing task, testSuite, test
+     * index numbers, and overall test number of the first failing test
+     * case. Returns null if no tests for the given tasks failed.
+     */
+    CodeEvalResult.prototype.getIndexOfFirstFailedTest = function(tasks) {
```

Comment:
This seems misnamed. The return value isn't an index.

That said, maybe it should be? The only actual call I can find to this just requires the overallTestNum. So maybe we should just return that?